core/bentley/src/BentleyError.ts

```diff
@@ -658,3 +658,60 @@ export class BentleyError extends Error {
     }
   }
 }
+
+function hasProperty<T extends string>(obj: object, propertyName: T): obj is { [key in T]: unknown } {
+  return propertyName in obj;
+}
+
+/** @internal */
+export function getErrorStack(error: unknown): string | undefined {
+  if (typeof error === "object" && error !== null && hasProperty(error, "stack") && typeof error.stack === "string")
+    return error.stack;
+
+  return undefined;
+}
+
+/** @internal */
+export function getErrorMessage(error: unknown): string | undefined {
+  if (typeof error === "string")
+    return error;
+
+  if (error instanceof Error)
+    return error.toString();
+
+  if (typeof error === "object") {
+    if (error === null)
+      return undefined;
+
+    if (hasProperty(error, "message") && typeof error.message === "string")
+      return error.message;
+
+    if (hasProperty(error, "msg") && typeof error.msg === "string")
+      return error.msg;
+
+    if (error.toString() !== "[object Object]")
+      return error.toString();
+  }
+
+  return undefined;
+}
+
+/** @internal */
+export function getErrorMetadata(error: unknown): { message: string, stack: string | undefined } {
```

Comment:
This is very confusing. It's called `getErrorMetadata`, but it's return type says it only returns the message/stack.  I don't think `getErrorMetadata` it should combine that into one object. The point of metadata on an exception object (I guess) is so you can log the metadata object separately.

Reply:
@kabentley I just took another crack at this, let me know what you think.

I'm hoping to follow this up with another PR that will hopefully simplify the metadata and exception handling in the @bentley/js-core logger...